CMake/macros.cmake

```diff
@@ -94,11 +94,11 @@ function(COPY_DIRECTORY_MAYBE_USING_SYMLINK SRC_DIR DST_DIR ${ARGN})
 endfunction()
 
 # Symlink or copy an individual file
-function(MAYBE_SYMLINK SRC_DIR DST_DIR)
+function(MAYBE_SYMLINK SRC_FILE DST_FILE)
   if(QMC_SYMLINK_TEST_FILES)
-    file(CREATE_LINK ${SRC_DIR} ${DST_DIR} SYMBOLIC)
+    file(CREATE_LINK ${SRC_FILE} ${DST_FILE} SYMBOLIC)
   else()
-    file(COPY ${SRC_DIR} DESTINATION ${DST_DIR})
+    configure_file(${SRC_FILE} ${DST_FILE} COPYONLY)
```

Comment:
Could you add a comment saying `configure_file` is used as a copy and explain why `file(copy` cannot be used. It is quite confusing when reading.

Reply:
Comment added. I also listed a few other options I considered but rejected because the require newer CMake versions.

May we set `-DQMC_SYMLINK_TEST_FILES=OFF` in a PR or nightly job to catch future errors?